src/dotnet/Audio.UI.Blazor/Components/AudioRecorder/AudioRecorder.cs

```diff
@@ -78,19 +69,14 @@ public async Task StartRecording(ChatId chatId, CancellationToken cancellationTo
         if (chatId.IsNone)
             throw new ArgumentOutOfRangeException(nameof(chatId));
 
-        var recordingCts = cancellationToken.CreateLinkedTokenSource();
         try {
-            _messageProcessor.Enqueue(new StartAudioRecorderCommand(chatId, recordingCts), recordingCts.Token);
-
-            await State.When(state => state?.IsRecording ?? false, recordingCts.Token)
-                .WaitAsync(TimeSpan.FromSeconds(5), recordingCts.Token);
+            var messageProcess = _messageProcessor.Enqueue(new StartAudioRecorderCommand(chatId), cancellationToken);
+            await messageProcess.WhenCompleted.ConfigureAwait(false);
```

Comment:
I am 80% sure it was intended to just enqueue the command here. It makes sense to look more carefully on what happens if you await for it here.

Reply:
Hmm, I see, it seems you switched from awaiting on State to awaiting for command completion.

Reply:
right, furthermore, command has its own execution timeout which was before my refactoring. It was 2 seconds, while this timeout 5s was sort of confusing and redundant.